omelet-core/src/main/java/omelet/mail/Email.java

```diff
@@ -41,340 +34,371 @@
-					}
-				}
-				break;
-			case TO:
-				for (Message msg : messages) {
-					for (Address addr : msg.getRecipients(RecipientType.TO)) {
-						if (addr.toString().contains(emailAddress)) {
-							returnMessage.add(msg);
-						}
-					}
-				}
-				break;
-			default:
-				break;
-			}
-		} catch (MessagingException e) {
-			// TODO Auto-generated catch block
-			LOGGER.error(e);
-		}
-		return null;
-	}
-
-	@Override
-	public List<Message> filerEmailsBySubject(List<Message> message,
-			String emailSubject) {
-		Stopwatch sw = new Stopwatch();
-		sw.start();
-		List<Message> returnMessage = new ArrayList<Message>();
-		LOGGER.info("Count of the message for filter by Subject"
-				+ message.size());
-		for (Message msg : message) {
-			try {
-				if (msg.getSubject().equalsIgnoreCase(emailSubject)) {
-					returnMessage.add(msg);
-				}
-			} catch (MessagingException e) {
-				// TODO Auto-generated catch block
-				LOGGER.error(e);
-			}
-		}
-		sw.stop();
-		LOGGER.info("Time Taken by Filter EmailBy Subjects is:"
-				+ sw.elapsedTime(TimeUnit.SECONDS));
-		return returnMessage;
-	}
-
-	@Override
-	public boolean checkPatternInEmail(Message message, String patterToMatch) {
-		// TODO Auto-generated method stub
-		String messageBody = getEmailBody(message);
-		return Pattern.matches(messageBody, patterToMatch);
-	}
+    final private String host;
+    final private String port;
+    final private String userName;
+    final private String password;
+    private String folderName;
+    private Folder folder;
+
+    private static final int maxcountEMailCheck = 10;
+    private static final Logger LOGGER = Logger.getLogger(Email.class);
+
+    private Email(Builder builder) {
+        MailProtocol protocol;
+
+        this.host = builder.host;
+        this.port = builder.port;
+        this.userName = builder.userName;
+        this.password = builder.password;
+        protocol = builder.protocol;
+        this.folderName = "Inbox";
+        connect(protocol);
+    }
+
+    private void connect(MailProtocol protocol) {
+        String hostPropertyKey;
+        String portPropertyKey;
+        String connectionTimeoutPropertyKey;
+        String timeoutPropertyKey;
+        Properties props = System.getProperties();
+        Session session;
+        switch (protocol) {
+            case POP3:
+                System.out.println("POP3");
+                hostPropertyKey = "mail.pop3.host";
+                portPropertyKey = "mail.pop3.port";
+                props.setProperty("mail.store.protocol", "pop3s");
+                props.setProperty("mail.pop3.user", userName);
+                connectionTimeoutPropertyKey = "mail.pop3.connectiontimeout";
+                timeoutPropertyKey = "mail.pop3.timeout";
+                props.setProperty("mail.pop3.socketFactory.class", "javax.net.ssl.SSLSocketFactory");
+                props.setProperty("mail.pop3.socketFactory.fallback", "false");
+                break;
+            case IMAP:
```

Comment:
should we make small private methods for configuring property for smtp,pop3, imap to increase readability , just a suggestion though